lib/utils/RecurringRidesGenerator.dart

```diff
@@ -1,151 +1,103 @@
-            .toList();
-        List<int> days = originalRide.recurringDays;
-
-        // find first occurrence
-        int daysUntilFirstOccurrence = days
-            .map((day) => daysUntilWeekday(originalRide.startTime, day))
-            .reduce(min);
-        DateTime rideStart = originalRide.startTime
-            .add(Duration(days: daysUntilFirstOccurrence));
-        int dayIndex = days.indexOf(rideStart.weekday);
-
-        // generate instances of recurring rides
-        while (rideStart.isBefore(originalRide.endDate) ||
-            rideStart.isAtSameMomentAs(originalRide.endDate)) {
-          // create the new ride and keep track of its parent
-          Ride rideInstance = Ride(
-              id: CarriageTheme.generatedRideID,
-              startLocation: originalRide.startLocation,
-              startAddress: originalRide.startAddress,
-              endLocation: originalRide.endLocation,
-              endAddress: originalRide.endAddress,
-              startTime: rideStart,
-              endTime: rideStart.add(rideDuration));
-          rideInstanceParentIDs[rideInstance] = originalRide.id;
-
-          // if all ride info is equal to a deleted ride, don't add it because it was edited
-          if (!wasDeleted(rideInstance, deletedInstances)) {
-            allRides.add(rideInstance);
-          }
+  /// Returns a list of all single-time rides and all future instances of repeating rides, based on [parentRides].
+  ///
+  /// Instances of repeating rides that do not exist yet use a temporary ID. Their parent rides, meaning the rides
+  /// in backend that they are generated from, will be used for editing instances of repeating rides that do not
+  /// exist in backend yet.
+  List<Ride> generateRecurringRides() {
+    Map<String, Ride> singleRidesByID = Map();
+    singleRides.forEach((singleRide) => singleRidesByID[singleRide.id] = singleRide);
+    List<Ride> generatedRides = [];
+
+    for (Ride originalRide in parentRides) {
+      DateTime origStart = originalRide.startTime;
+      Duration rideDuration = originalRide.endTime.difference(origStart);
+      List<int> days = originalRide.recurringDays;
+
+      // find first occurrence
+      DateTime now = DateTime.now();
+      DateTime rideCreationTime = DateTime(now.year, now.month, now.day, 10, 5);
+
+      DateTime today = DateTime(now.year, now.month, now.day, origStart.hour, origStart.minute);
+      DateTime firstPossibleDate = origStart.isBefore(today) ? today : originalRide.startTime;
+      int daysUntilFirstOccurrence = days.map((day) => daysUntilWeekday(firstPossibleDate, day)).reduce(min);
+      DateTime rideStart = firstPossibleDate.add(Duration(days: daysUntilFirstOccurrence));
+      int dayIndex = days.indexOf(rideStart.weekday);
+
+      // generate instances of recurring rides
+      while (rideStart.isBefore(originalRide.endDate) || rideStart.isAtSameMomentAs(originalRide.endDate)) {
+        Ride rideInstance = Ride(
```

Comment:
Do we have support for deleting recurring rides in the backend yet?

Reply:
Yep, Simran's edit recurring rides PR just got merged so I'll be working on this today!